lib/scala/semver/src/main/java/org/enso/semver/SemVer.java

```diff
@@ -0,0 +1,145 @@
+package org.enso.semver;
+
+import java.util.Optional;
+import java.util.regex.Pattern;
+
+import scala.util.Failure;
+import scala.util.Success;
+import scala.util.Try;
+
+/**
+ * A facade hiding the implementation of the Semantic Versioning used in Enso.
+ */
+public class SemVer {
+
+    private long major;
+    private long minor;
+    private long patch;
+    private Optional<String> preRelease;
+    private Optional<String> prefix;
+
+    private static final Pattern regex = Pattern.compile("([a-z]+-)?(\\d+)\\.(\\d+)\\.(\\d+)(-[a-z]+)?");
```

Comment:
btw. why do we allow the prefix? When was that used?

Reply:
Also note that we are ignoring the build suffix (after a `+`) that is [technically part of Semver](https://semver.org/#backusnaur-form-grammar-for-valid-semver-versions). But I guess that's ok since we are not using those.

Reply:
A number of our tests seem to have `enso-x.y.z-<release>` pattern. And yes, I'm ignoring `+` on purpose, this is highly Enso-specific.